src/components/SearchBar/SearchBar.tsx

```diff
@@ -1,52 +1,43 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { SearchIcon } from '../../assets/icons';
 import './SearchBar.scss';
 
 interface SearchBarProps {
   onSearch: (searchValue: string) => void;
 }
 
-interface SearchBarState {
-  value: string;
-}
+function SearchBar(props: SearchBarProps) {
+  const [searchValue, setSearchValue] = useState(localStorage.getItem('searchValue') || '');
 
-class SearchBar extends React.Component<SearchBarProps, SearchBarState> {
-  constructor(props: SearchBarProps) {
-    super(props);
-    this.state = {
-      value: localStorage.getItem('searchValue') || '',
+  useEffect(() => {
+    const handleBeforeUnload = () => {
+      localStorage.setItem('searchValue', searchValue);
     };
-  }
-
-  componentDidMount() {
-    window.addEventListener('beforeunload', this.handleBeforeUnload);
-  }
 
-  componentWillUnmount() {
-    window.removeEventListener('beforeunload', this.handleBeforeUnload);
-    localStorage.setItem('searchValue', this.state.value);
-  }
+    window.addEventListener('beforeunload', handleBeforeUnload);
 
-  handleBeforeUnload = () => {
-    localStorage.setItem('searchValue', this.state.value);
-  };
+    return () => {
+      window.removeEventListener('beforeunload', handleBeforeUnload);
+      localStorage.setItem('searchValue', searchValue);
+    };
+  }, [searchValue]);
```

Comment:
This hook will be triggered every time you enter a character, I think it's not optimal!